mmstudio/src/main/java/org/micromanager/data/internal/multipagetiff/MultipageTiffReader.java

```diff
@@ -504,16 +582,22 @@ private Image readImage(IFDData data) throws IOException {
          // a nio buffer directly as the Image storage (even better if memory
          // mapped).
          switch (pixelType) {
-            case GRAY8:
-               return new DefaultImage(pixelBuffer.array(), formatPmap,
+            case GRAY8: {
+               Image img = new DefaultImage(pixelBuffer.array(), formatPmap,
                      coords, metadata);
```

Comment:
`ByteBuffer.array()` only works if the byte buffer is backed by a Java array (i.e., it's not a direct byte buffer) and returns the backing array.

And indeed the `DefaultImage` constructor (re-)wraps the array as a buffer, which would mean that immediately recycling the buffer is incorrect.

Reply:
I'm a little confused here -- if the previous implementation was reusing a single buffer (per reader), was that also overwriting the contents of `DefaultImage`s that are already passed out???

Reply:
Similarly confused.  This situation only applied to 8 bit images, which probably were not used much.  To be safe, I now clone the ByteBuffer (which - as far as I understand - should make a copy of the values in the array).

Reply:
Turns out that PixelBuffer is not backed by an array, so that code failed.  I added code to check if the array backing exists and make a copy if it does not.  That makes the 8 bit situation more similar to the 16 bit case.  Seems to work now.

Reply:
Also remove functions/class added in #1934, that are indeed no longer needed.